website/server/models/group.js

```diff
@@ -1410,6 +1410,7 @@ schema.methods.leave = async function leaveGroup (user, keep = 'keep-all', keepC
     // double check the member count is correct
     // so we don't accidentally delete a group that still has users in it
     let members;
+    this.memberCount = group.getMemberCount();
```

Comment:
Good catch here as well but at this should be `await group.getMemberCount()` and also the lines belowe (1414-1420) aren't needed anymore now, could all be substituted by:

```
const memberCount = await group.getMemberCount();
if ((memberCount - 1) === 0) { // -1 to make sure the user leaving is not counted
    promises.push(group.remove());
    return Promise.all(promises);
}
```

Reply:
Thanks for the comments!

I removed the lines that you indicated. Those lines also included code that contained the list of members that need to be deleted.

Four test cases are failing due to the newly pushed changes. I tried to figure out the reason and I believe that the members are not being removed from a group. The line `_.remove(members, { _id: user._id });` was instrumental in removing the members but we are currently not doing so, as far as my understanding is concerned. This is what we are currently doing.

https://github.com/HabitRPG/habitica/blob/abe60489fd77875292e50729679dca921c2ba9bc/website/server/models/group.js#L1414-L1417

Hence, the test cases are failing. Can you please help me out with this one?